third_party/airflow/armada/operators/armada.py

```diff
@@ -102,3 +104,23 @@ def execute(self, context) -> None:
             "Armada Job finished with %s and message: %s", job_state, job_message
         )
         airflow_error(job_state, self.name, job_id)
+
+
+def annotate_job_request_items(context, job_request_items):
+    """
+    Annotates the inbound job request items with the context task ID
+
+    :param context: The airflow context.
+
+    :param job_request_items: The job request items we plan to send to armada
+
+    :return: annotated job request items for armada
+    """
+    task_instance = context["ti"]
+    task_id = task_instance.task_id
+
+    for item_list in job_request_items:
+        for item in item_list:
+            item.annotations["armadaproject.io/taskId"] = task_id
```

Comment:
Should annotations be stored as constants? Is there a spec somewhere for them?

Reply:
re: spec, there is only the ticket (#1608) AFAIK. 

re: const -- i usually think that a const is unecessary for singular usage, but I moved this to a const var.